soh/soh/Enhancements/randomizer/3drando/fill.cpp

```diff
@@ -485,20 +485,18 @@ static void GeneratePlaythrough() {
     GetAccessibleLocations(ctx->allLocations, SearchMode::GeneratePlaythrough);
 }
 
-RandomizerArea LookForExternalArea(Area* curRegion, std::vector<RandomizerRegion> alreadyChecked){//RANDOTODO curREGION
-  for (auto& entrance : curRegion->entrances) {
+RandomizerArea LookForExternalArea(const Area* const currentRegion, std::vector<RandomizerRegion> &alreadyChecked){
+  for (const auto& entrance : currentRegion->entrances) {
     RandomizerArea otherArea = entrance->GetParentRegion()->GetArea();
-    if (otherArea != RA_LINKS_POCKET){ //if it's links pocket, do not propogate this, link's pocket is not a real Area
-      if(otherArea != RA_NONE){
-        return otherArea;
-        //if the area hasn't already been checked, check it
-      } else if (std::find(alreadyChecked.begin(), alreadyChecked.end(), entrance->GetParentRegionKey()) == alreadyChecked.end()) {
+    if(otherArea == RA_NONE && std::find(alreadyChecked.begin(), alreadyChecked.end(), entrance->GetParentRegionKey()) == alreadyChecked.end()){
+        //if the region is in RA_NONE and hasn't already been checked, check it
         alreadyChecked.push_back(entrance->GetParentRegionKey());
-        RandomizerArea passdown = LookForExternalArea(entrance->GetParentRegion(), alreadyChecked);
+        const RandomizerArea passdown = LookForExternalArea(entrance->GetParentRegion(), alreadyChecked);
         if(passdown != RA_NONE){
           return passdown;
         }
-      }
+    } else if (otherArea != RA_LINKS_POCKET){ //if it's links pocket, do not propogate this, Link's Pocket is not a real Area
```

Comment:
This "else if" is unnecessary, you can just write` if (otherArea != RA_LINKS_POCKET)`

Reply:
The compiler will add it, but matching the compiler result sounds good for clarity of intent when all it costs is 5 characters, give or take a newline.